jlm/llvm/ir/operators/lambda.hpp

```diff
@@ -401,302 +423,9 @@ class node final : public jlm::rvsdg::structural_node
-      before any context variable was added.
-    */
-    return lambda->nfctarguments() > index + 1 ? lambda->fctargument(index + 1) : nullptr;
-  }
-};
-
-/** \brief Lambda context variable argument
- */
-class cvargument final : public rvsdg::RegionArgument
-{
-  friend ::jlm::llvm::lambda::node;
-
-public:
-  ~cvargument() override;
-
-  cvargument &
-  Copy(rvsdg::Region & region, jlm::rvsdg::structural_input * input) override;
-
-private:
-  cvargument(rvsdg::Region * region, cvinput * input)
-      : rvsdg::RegionArgument(region, input, input->Type())
-  {}
-
-  static cvargument *
-  create(rvsdg::Region * region, lambda::cvinput * input)
-  {
-    auto argument = new cvargument(region, input);
-    region->append_argument(argument);
-    return argument;
-  }
-
-public:
-  cvinput *
-  input() const noexcept
-  {
-    return jlm::util::AssertedCast<cvinput>(rvsdg::RegionArgument::input());
-  }
-};
-
-/** \brief Lambda result
- */
-class result final : public rvsdg::RegionResult
-{
-  friend ::jlm::llvm::lambda::node;
-
-public:
-  ~result() override;
-
-  result &
-  Copy(rvsdg::output & origin, jlm::rvsdg::structural_output * output) override;
-
-private:
-  explicit result(jlm::rvsdg::output * origin)
-      : rvsdg::RegionResult(origin->region(), origin, nullptr, origin->Type())
-  {}
-
-  static result *
-  create(jlm::rvsdg::output * origin)
-  {
-    auto result = new lambda::result(origin);
-    origin->region()->append_result(result);
-    return result;
-  }
-
-public:
-  lambda::output *
-  output() const noexcept
-  {
-    return jlm::util::AssertedCast<lambda::output>(rvsdg::RegionResult::output());
-  }
-};
-
-/** \brief Lambda result iterator
- */
-class node::fctresiterator final : public jlm::rvsdg::input::iterator<lambda::result>
-{
-  friend ::jlm::llvm::lambda::node;
-
-  constexpr explicit fctresiterator(lambda::result * result)
-      : jlm::rvsdg::input::iterator<lambda::result>(result)
-  {}
-
-  [[nodiscard]] lambda::result *
-  next() const override
-  {
-    auto index = value()->index();
-    auto lambda = jlm::util::AssertedCast<lambda::node>(value()->region()->node());
-
-    return lambda->nfctresults() > index + 1 ? lambda->fctresult(index + 1) : nullptr;
-  }
-};
-
-/** \brief Lambda result const iterator
- */
-class node::fctresconstiterator final : public jlm::rvsdg::input::constiterator<lambda::result>
-{
-  friend ::jlm::llvm::lambda::node;
-
-  constexpr explicit fctresconstiterator(const lambda::result * result)
-      : jlm::rvsdg::input::constiterator<lambda::result>(result)
-  {}
-
-  [[nodiscard]] const lambda::result *
-  next() const override
-  {
-    auto index = value()->index();
-    auto lambda = jlm::util::AssertedCast<lambda::node>(value()->region()->node());
-
-    return lambda->nfctresults() > index + 1 ? lambda->fctresult(index + 1) : nullptr;
-  }
+  std::vector<jlm::llvm::attributeset> argumentAttributes_;
```

Comment:
`ArgumentAttributes_`

Reply:
now I am confused -- I thought functions and classes UpperCamelCase and variables lowerCamelCase, no?

Reply:
Function and Classes: UpperCamelCase
Local Variables: lowerCamelCase
Class Attributes: UpperCamelCaseWith_

Reply:
renamed